src/main/java/core/basesyntax/BalanceOperation.java

```diff
@@ -0,0 +1,14 @@
+import java.util.Map;
+
+public class BalanceOperation implements OperationHandler {
+    private final Map<String, Integer> fruitStorage;
+
+    public BalanceOperation(Map<String, Integer> fruitStorage) {
+        this.fruitStorage = fruitStorage;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        fruitStorage.put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
Please ensure that the balance is not negative before updating the storage. If the quantity is negative, a `RuntimeException` should be thrown to prevent incorrect data in the storage.